torch/csrc/jit/passes/symbolic_shape_runtime_fusion.cpp

```diff
@@ -376,5 +389,31 @@ RegisterOperators reg_guard({
         },
         aliasAnalysisFromSchema()),
 });
+
+Operation createTensorExprDynamicGroup(const Node* node) {
+  auto graph = node->g(attr::Subgraph);
+  // Code code(graph, "");
+  // InterpreterState interpreter{code};
+  // return [=](Stack& stack) mutable {
+  //   std::cout << "In TensorExprDynamicGroup execution, stack size = " << stack.size() << std::endl;
+  //   interpreter.run(stack);
+  //   return 0;
+  // };
+  return [=](Stack& stack) {
```

Comment:
nit: maybe be explicit of things we are copying

Reply:
wouldnt this be instantiated a code object/Interpreter state on every invocation of the op? i think we want to create an InterpreterState outside of the lambda

Reply:
If I create the `InterpreterState` outside the lambda, then the context, specifically `frames` in `InterpreterStateImpl` is garbled when the lambda is getting executed. That is the reason I had to move the instantiation to inside lambda. 

If there is a way to instantiate that outside and reuse it for every invocation of the op, that would be awesome. And would definitely help perf

Reply:
@navahgar `InterpreterState` is pretty lightweight or used to be. On the other hand if you can cache `Code` object construction, that could be pretty significant savings since `Code` translates JIT graphs into interpreter instructions, does register allocation and a bunch of other passes. Both, legacy and profiling executors, cache `Code` objects in `ExecutionPlan` for this exact reason.

You could do 

```cpp
Code code(graph, ""); // code should be copyable.
return [code](Stack& stack) {
```

`InterpreterState` should be doing mainly two things: 
* push the Code object on the `frames` stack
* Reserve enough spaces for virtual registers

Both should be fairly low-overhead. Not sure it would be worthwhile to reuse the `InterpreterState` vs `Code` tbh

cc: @eellison

Reply:
This is supposed to be an optimized kernel execution... so even small overheads matter here. Would be great to find a way to avoid the duplicate instantiation